scalableminds/webknossos
# ND data loading with other axis orders

- enables ND data loading for all data formats with AxiOrder and AdditionalCoordinates (precomputed, n5, zarr3) (previously only zarr2)
 - volume annotation data.zips should now be viewable as datasets after unpacking (fixed the indices of their additional coordinates)
 - unifies DatasetArrayHandle (previously CubeHandle, no longer a trait with multiple implementations)
 - no more distinction between reading with or without AdditionalCoordinates from DatasetArray
 - Added some more tryos for better error handling
 - FullAxisOrder is now constructed for reading with arbitrary ND axis orders, with permutations from array order to wk order (tcxyz), including one for flipping c to fortran in the process
 - Adds some tests for these permutatinos

### Steps to test:
- Unpack zarr volume download data zip (both 3d and 4d), should be viewable as dataset
- View 4d and 3d and 2d zarr datasets, should be viewable

### TODOs:
- [x] fix volume annotation metadata (additional axes have wrong indices)
- [x] fix permute (needs to take additional axes into account)
- [x] test with volume annotations
- [x] test with 2d zarr
- [x] add tests for the permutations in c and f
- [x] fix viewing nd datasets *without* the additional coordinates in the json (cxyz should still be accessible if normal axisOrder field is correct)

### Issues:
- fixes [#7280](https://github.com/scalableminds/webknossos/issues/7280)

------
